pkg/rpc/client.go

```diff
@@ -64,7 +70,7 @@ func (c *Client) listen() {
 	for {
```

Comment:
Since we seem to be passing the rpc clients around with interfaces, we can probably get rid of the listening goroutine.

Reply:
Agreed, the indirection looks unnecessary